android/src/main/java/com/aerobotics/DjiMobile/CameraEventDelegate.java

```diff
@@ -34,40 +34,49 @@ public void sendEvent(Object payload) {
 
   private CameraEventObservable cameraEventObservable = new CameraEventObservable();
 
-  private CameraEventDelegate cameraDelegateSenderInstance = this;
-
-  private boolean isProductConnected = false;
   private boolean isCameraConnected = false;
-
-  private KeyListener ProductConnectedKeyListener = new KeyListener() {
-    @Override
-    public void onValueChange(@Nullable Object oldValue, @Nullable Object newValue) {
-      if (newValue != null && newValue instanceof Boolean) {
-        isProductConnected = (boolean)newValue;
-        setCameraCallbacks();
-      }
-    }
-  };
+  private boolean callbacksSet = false;
 
   private KeyListener CameraConnectedKeyListener = new KeyListener() {
     @Override
     public void onValueChange(@Nullable Object oldValue, @Nullable Object newValue) {
       if (newValue != null && newValue instanceof Boolean) {
-        isCameraConnected = (boolean)newValue;
-        setCameraCallbacks();
+        if ((boolean) newValue && !isCameraConnected) {
```

Comment:
There's no problem with a whitespace between `(boolean) newValue` is there?

Reply:
No issue other than inconsistent styling